tests/unit/r2i/engine.cc

```diff
@@ -0,0 +1,96 @@
+/* Copyright (C) 2020 RidgeRun, LLC (http://www.ridgerun.com)
+ * All Rights Reserved.
+ *
+ * The contents of this software are proprietary and confidential to RidgeRun,
+ * LLC.  No part of this program may be photocopied, reproduced or translated
+ * into another programming language without prior written consent of
+ * RidgeRun, LLC.  The user is free to modify the source code after obtaining
+ * a software license from RidgeRun.  All source code changes must be provided
+ * back to RidgeRun without any encumbrance.
+*/
+
+#include <r2i/engine.h>
+
+#include <CppUTest/CommandLineTestRunner.h>
+#include <CppUTest/MemoryLeakDetectorNewMacros.h>
+#include <CppUTest/MemoryLeakDetectorMallocMacros.h>
+#include <CppUTest/TestHarness.h>
+
+namespace mock {
+class Preprocessing : public r2i::IPreprocessing {
+ public:
+  Preprocessing () {}
+  r2i::RuntimeError apply(r2i::IFrame &data) {
+    return r2i::RuntimeError();
```

Comment:
How are you testing the handling of a preprocessing fails?

Reply:
This is being tested in the tests for Engine::Predict() in the different backend, this is since Preprocess() and Postprocess() are protected member being called by Engine::Predict().